cluster/gce/gci/audit_policy_test.go

```diff
@@ -0,0 +1,260 @@
+		ingress             = userinfo("system:unsecured", user.AllAuthenticated, user.SystemPrivilegedGroup)
+		kubelet             = userinfo("kubelet", user.AllAuthenticated, user.NodesGroup)
+		node                = userinfo("system:node:node-123", user.AllAuthenticated, user.NodesGroup)
+		controller          = userinfo(user.KubeControllerManager, user.AllAuthenticated)
+		scheduler           = userinfo(user.KubeScheduler, user.AllAuthenticated)
+		apiserver           = userinfo(user.APIServerUser, user.SystemPrivilegedGroup)
+		autoscaler          = userinfo("cluster-autoscaler", user.AllAuthenticated)
+		npd                 = userinfo("system:node-problem-detector", user.AllAuthenticated)
+		npdSA               = serviceaccount.UserInfo("kube-system", "node-problem-detector", "")
+		namespaceController = serviceaccount.UserInfo("kube-system", "namespace-controller", "")
+		endpointController  = serviceaccount.UserInfo("kube-system", "endpoint-controller", "")
+		defaultSA           = serviceaccount.UserInfo("default", "default", "")
+
+		allUsers = []user.Info{anonymous, kubeproxy, ingress, kubelet, node, controller, scheduler, apiserver, autoscaler, npd, npdSA, namespaceController, endpointController, defaultSA}
+	)
+
+	// Resources for test cases
+	var (
+		nodes           = resource("nodes")
+		nodeStatus      = resource("nodes", "", "", "status")
+		endpoints       = resource("endpoints", "default")
+		sysEndpoints    = resource("endpoints", "kube-system")
+		services        = resource("services", "default")
+		serviceStatus   = resource("services", "default", "", "status")
+		configmaps      = resource("configmaps", "default")
+		sysConfigmaps   = resource("configmaps", "kube-system")
+		namespaces      = resource("namespaces")
+		namespaceStatus = resource("namespaces", "", "", "status")
+		namespaceFinal  = resource("namespaces", "", "", "finalize")
+		podMetrics      = resource("podmetrics", "default", "metrics.k8s.io")
+		nodeMetrics     = resource("nodemetrics", "", "metrics.k8s.io")
+		pods            = resource("pods", "default")
+		podStatus       = resource("pods", "default", "", "status")
+		secrets         = resource("secrets", "default")
+		tokenReviews    = resource("tokenreviews", "", "authentication.k8s.io")
+		deployments     = resource("deployments", "default", "apps")
+		clusterRoles    = resource("clusterroles", "", "rbac.authorization.k8s.io")
+		events          = resource("events", "default")
+		foobars         = resource("foos", "default", "example.com")
+		foobarbaz       = resource("foos", "default", "example.com", "baz")
+	)
+
+	// Aliases
+	const (
+		none     = audit.LevelNone
+		metadata = audit.LevelMetadata
+		request  = audit.LevelRequest
+		response = audit.LevelRequestResponse
+	)
+
+	at := auditTester{
+		T:       t,
+		checker: auditpolicy.NewChecker(policy),
+	}
+
+	at.testResources(none, kubeproxy, "watch", endpoints, sysEndpoints, services, serviceStatus)
```

Comment:
Any reason to not make this a table test?

Reply:
If I lay these all out as structs, I think it would look like:
```
{
  level: none,
  users: []user.Info{kubeproxy},
  verbs: []string{"watch"},
  resources: []Resource{endpoints, sysEndpoints, services, serviceStatus},
}
```
Which is much more verbose. I could create a testcase geneartor method using a similar technique to what I have now, but IMO that just adds unnecessary complexity.